server/services/transport/small-webrtc.mdx

```diff
@@ -0,0 +1,454 @@
+
+```python
+from pipecat.audio.vad.silero import SileroVADAnalyzer
+from pipecat.pipeline.pipeline import Pipeline
+from pipecat.transports.base_transport import TransportParams
+from pipecat.transports.network.small_webrtc import SmallWebRTCTransport
+
+async def run_bot(webrtc_connection):
+    # Create the WebRTC transport with the provided connection
+    transport = SmallWebRTCTransport(
+        webrtc_connection=webrtc_connection,
+        params=TransportParams(
+            audio_in_enabled=True,   # Accept audio from the client
+            audio_out_enabled=True,  # Send audio to the client
+            vad_enabled=True,        # Enable Voice Activity Detection
+            vad_analyzer=SileroVADAnalyzer(),
+            vad_audio_passthrough=True,
+        ),
+    )
+
+    # Set up your services and context
+
+    # Create the pipeline
+    pipeline = Pipeline([
+        transport.input(),              # Receive audio from client
+        stt,                            # Convert speech to text
+        context_aggregator.user(),      # Add user messages to context
+        llm,                            # Process text with LLM
+        tts,                            # Convert text to speech
+        transport.output(),             # Send audio responses to client
+        context_aggregator.assistant(), # Add assistant responses to context
+    ])
+
+    # Register event handlers
+    @transport.event_handler("on_client_connected")
+    async def on_client_connected(transport, client):
+        logger.info("Client connected")
+        # Start the conversation when client connects
+        await task.queue_frames([context_aggregator.user().get_context_frame()])
+
+    @transport.event_handler("on_client_disconnected")
+    async def on_client_disconnected(transport, client):
+        logger.info("Client disconnected")
+
+    @transport.event_handler("on_client_closed")
+    async def on_client_closed(transport, client):
+        logger.info("Client closed")
+        await task.cancel()
+```
+
+## Examples
+
+To see a complete implementation, check out the following examples:
+
+<CardGroup>
+  <Card
+    title="Video Transform"
+    icon="camera"
+    href="https://github.com/pipecat-ai/pipecat/tree/main/examples/p2p-webrtc/video-transform"
+  >
+    Demonstrates real-time video processing using WebRTC transport
+  </Card>
+  <Card
+    title="Voice Agent"
+    icon="microphone"
+    href="https://github.com/pipecat-ai/pipecat/tree/main/examples/p2p-webrtc/voice-agent"
+  >
+    Implements a voice assistant using WebRTC for audio communication
+  </Card>
+</CardGroup>
+
+## Media Handling
+
+### Audio
+
+Audio is processed in 20ms chunks by default. The transport handles audio format conversion and resampling as needed:
+
+- Input audio is processed at 16kHz (mono) to be compatible with speech recognition services
+- Output audio can be configured to match your application's requirements
```

Comment:
We can configure the `sample_rate`, but there are some limitations regarding the output:
- The audio we send is always mono.
- It always assumes 16-bit PCM, which is WebRTC's expected format.